### PR TITLE
complete task 12-including-comment-count

### PR DESCRIPTION
**Pull request by IndyS1ngh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -74,6 +74,7 @@ describe("GET /api/articles/:article_id", () => {
           votes: 100,
           article_img_url:
             "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+          comment_count: 11,
         });
```
Comment (nc-mick-fay) [inline __tests__/app.test.js:77]: When following Test Driven Development principles, it's not good practice to go back and change old tests. Preferably you would write an entire new test to check that comment_count is added to your responses

File: models/models.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ exports.selectEndpoints = () => {
 
 exports.selectArticleById = (article_id) => {
   return db
-    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .query(`SELECT articles.*, (COUNT(comments.article_id) :: INTEGER) AS comment_count FROM comments RIGHT JOIN articles ON articles.article_id = comments.article_id WHERE articles.article_id = $1 GROUP BY articles.article_id;`, [article_id])
     .then(({ rows }) => {
       if (!rows.length) {
         return Promise.reject({ status: 404, msg: "article does not exist" });
```
Comment (nc-mick-fay) [inline models/models.js:20]: Nice changes to your query here!

**Review by nc-mick-fay — APPROVED** [review]
Great changes to your query! This is ready to merge into your main branch.

Remember when following TDD it is bad practice to go back and change tests. It is preferable for you to build an entire new test to check that comment_count gets included into your response object.
Your tests are a source of documentation. Adding a new test for a new feature shows how your code has developed - squeezing one line into an old test could easily be overlooked!

**Review by Slyose — APPROVED** [review]
Looks good! Great response to feedback, feel free to merge ✅

**Pull request merged**